apps/meteor/client/views/omnichannel/departments/DepartmentsRoute.js

```diff
@@ -67,7 +72,23 @@ function DepartmentsRoute() {
 			}),
 	);
 
-	const { value: data = {}, reload } = useEndpointData('/v1/livechat/department', { params: query });
+	const query = {
+		onlyMyDepartments,
+		text: debouncedParams.text,
+		offset: debouncedParams.current,
+		count: debouncedParams.itemsPerPage,
+		sort: JSON.stringify({
+			[debouncedSort.column]: sortDir(debouncedSort.direction),
+			usernames: debouncedSort.column === 'name' ? sortDir(debouncedSort.direction) : undefined,
+		}),
+		fields: JSON.stringify({ name: 1, username: 1, emails: 1, avatarETag: 1 }),
+	};
+
+	const getDepartments = useEndpoint('GET', '/v1/livechat/department');
+
+	const { data, refetch } = useQuery(['getDepartments', query], async () => getDepartments(query));
```

Comment:
Use a URI-like style for query keys:

```suggestion
	const { data, refetch } = useQuery(['omnichannel', 'departments', query], async () => getDepartments(query));
```